@xen-orchestra/lite/src/App/TreeView.tsx

```diff
@@ -119,12 +140,12 @@ const TreeView = withState<State, Props, Effects, Computed, ParentState, ParentE
         ),
     },
   },
-  ({ state, defaultSelectedNodes }) =>
-    state.collection === undefined ? null : (
+  ({ state: { collection, selectedNodes }, effects }) =>
+    collection === undefined ? null : (
       <div style={{ padding: '10px' }}>
-        <Tree collection={state.collection} defaultSelectedNodes={defaultSelectedNodes} />
+        <Tree collection={collection} selectedNodes={selectedNodes} onNodeSelect={effects.setSelectedNodes} />
```

Comment:
```suggestion
        <Tree
          collection={collection}
          selectedNodes={objectId !== undefined ? _selectedNodes : []}
          onNodeSelect={effects.setSelectedNodes}
        />
```

To remove  `this.props.history.push('/infrastructure/type/' + nodeIds[0])` from `setSelectedNodes`, what do you think?

Reply:
It won't work.
The URL, is updated in case that a node is selected.
So if we remove the `history.push` `selectedNodes` will alway be `undefined`